src/AudioFileReader.h

```diff
@@ -40,6 +53,8 @@ class AudioFileReader
         virtual bool open(const char* input_filename, bool show_info = true) = 0;
 
         virtual bool run(AudioProcessor& processor) = 0;
+
+        virtual bool configure(int channels, int bits_per_sample, int rate, Signedness s, Endianness e);
```

Comment:
Sorry, I think the example I gave before maybe was a bit misleading. This function only needs to be in `SndFileAudioFileReader`, not the base class, so it also doesn't need to be `virtual`. Then we'd write:

```cpp
SndFileAudioFileReader* sndfile_audio_file_reader = new SndFileAudioFileReader;
sndfile_audio_file_reader->configure(...);
reader.reset(sndfile_audio_file_reader);
```

Reply:
Ah right, I thought about that, but didn't think of creating and configuring the instance first, before resetting the pointer :facepalm: Have not touched c++ code in a while...

Reply:
Done.

Reply:
done